src/Tribe/Attendee_Repository.php

```diff
@@ -663,4 +693,433 @@ protected function limit_list( $key, $list ) {
+	 *
+	 * @since TBD
+	 *
+	 * @param Tribe__Tickets__Ticket_Object $ticket        The ticket object.
+	 * @param array                         $attendee_data List of additional attendee data.
+	 *
+	 * @return WP_Post|false The new post object or false if unsuccessful.
+	 *
+	 * @throws Tribe__Repository__Usage_Error If the argument types are not set as expected.
+	 */
+	public function create_attendee_for_ticket( $ticket, $attendee_data ) {
+		// Require the ticket be a ticket object.
+		if ( ! $ticket instanceof Tribe__Tickets__Ticket_Object ) {
+			throw new Tribe__Repository__Usage_Error( 'You must provide a ticket object when creating an attendee from the Attendees Repository class' );
+		}
+
+		// Set the attendee arguments accordingly.
+		$this->set_attendee_args( $attendee_data, $ticket );
+
+		// Update the attendee data for referencing with what we handled in the set_attendee_args().
+		$attendee_data = array_merge( $attendee_data, $this->updates );
+
+		// Create the new attendee.
+		$attendee = $this->create();
+
+		// Handle any further attendee updates.
+		$this->save_extra_attendee_data( $attendee, $attendee_data, $ticket );
+
+		// Trigger creation actions.
+		$this->trigger_create_actions( $attendee, $attendee_data, $ticket );
+
+		// Clear the attendee cache if post_id is provided.
+		if ( ! empty( $this->updates['post_id'] ) ) {
+			$this->attendee_provider->clear_attendees_cache( $this->updates['post_id'] );
+		}
+
+		return $attendee;
+	}
+
+	/**
+	 * Create an attendee object from ticket and attendee data.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $attendee_data  List of attendee data to be saved.
+	 * @param bool  $return_promise Whether to return a promise object or just the ids
+	 *                              of the updated posts; if `true` then a promise will
+	 *                              be returned whether the update is happening in background
+	 *                              or not.
+	 *
+	 * @return array|Tribe__Promise A list of the post IDs that have been (synchronous) or will
+	 *                              be (asynchronous) updated if `$return_promise` is set to `false`;
+	 *                              the Promise object if `$return_promise` is set to `true`.
+	 *
+	 * @throws Tribe__Repository__Usage_Error If the argument types are not set as expected.
+	 */
+	public function update_attendee( $attendee_data, $return_promise = false ) {
+		if ( empty( $attendee_data['attendee_id'] ) ) {
+			throw new Tribe__Repository__Usage_Error( 'You must provide the attendee_id when updating an attendee from the Attendees Repository class' );
+		}
+
+		$this->by( 'id', $attendee_data['attendee_id'] );
+
+		// Set the attendee arguments accordingly.
+		$this->set_attendee_args( $attendee_data );
+
+		// Update the attendee.
+		$saved = $this->save( $return_promise );
+
+		if ( $return_promise ) {
+			$repository = $this;
+
+			return $saved->then( static function() use ( $repository, $attendee_data ) {
```

Comment:
:no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line

<sub>hash: 1b81defd4d6f628596088fb55bc628e7</sub>